server/src/main/java/com/paypal/selion/proxy/SeLionSauceProxy.java

```diff
@@ -32,62 +33,84 @@
 import org.openqa.grid.internal.BaseRemoteProxy;
 import org.openqa.grid.internal.Registry;
 import org.openqa.grid.internal.TestSession;
+import org.openqa.grid.selenium.proxy.DefaultRemoteProxy;
 
 import com.paypal.selion.logging.SeLionGridLogger;
 
 /**
  * This customized version of {@link BaseRemoteProxy} basically helps redirect all traffic to the SauceLabs cloud.
  */
-public class SeLionSauceProxy extends BaseRemoteProxy {
+public class SeLionSauceProxy extends DefaultRemoteProxy {
 
     private static final SeLionGridLogger LOGGER = SeLionGridLogger.getLogger(SeLionSauceProxy.class);
+
+    private static final String DEFAULT_TO_USERNAME_TOKEN = "$username";
+    private static final String PARENT_TUNNEL_CAPABILITY = "parent-tunnel";
+    private static final String USERNAME_CAPABILITY = "username";
+
     private final SauceLabsRestApi sauceApi;
+    private final SauceConfigReader reader;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'reader' could be replaced by a local variable.](https://www.codacy.com/app/SeLion/paypal-SeLion/file/2405043082/issues/source?bid=2928425&fileBranchId=3233060#l52)